test/cadet_web/controllers/assessments_controller_test.exs

```diff
@@ -367,6 +369,48 @@ defmodule CadetWeb.AssessmentsControllerTest do
       end
     end
 
+    test "it renders xp, grade for students", %{
+      conn: conn,
+      users: users,
+      assessments: assessments
+    } do
+      for role <- Role.__enum_map__() do
+        user = Map.get(users, role)
+
+        for {_type,
+             %{
+               assessment: assessment,
+               mcq_answers: [mcq_answers | _],
+               programming_answers: [programming_answers | _]
+             }} <- assessments do
+          expected =
+            if role == :student do
+              Enum.map(
+                programming_answers ++ mcq_answers,
+                &%{
+                  "xp" => &1.xp + &1.xp_adjustment,
+                  "grade" => &1.grade + &1.adjustment
+                }
+              )
+            else
+              fn -> %{"xp" => 0, "grade" => 0} end
+              |> Stream.repeatedly()
+              |> Enum.take(length(programming_answers) + length(mcq_answers))
+            end
+
+          resp =
+            conn
+            |> sign_in(user)
+            |> get(build_url(assessment.id))
+            |> json_response(200)
+            |> Map.get("questions", [])
+            |> Enum.map(&%{"xp" => &1["xp"], "grade" => &1["grade"]})
```

Comment:
how about `Enum.map(&Map.take(&1, "xp", "grade"))`

Reply:
OH omg this is what I've been looking for!

Reply:
Done